src/test/ui/consts/dangling-alloc-id-ice.rs

```diff
@@ -1,11 +1,13 @@
 // https://github.com/rust-lang/rust/issues/55223
+#![allow(const_err)]
 
 union Foo<'a> {
     y: &'a (),
     long_live_the_unit: &'static (),
 }
 
-const FOO: &() = { //~ ERROR any use of this value will cause an error
+const FOO: &() = { //~ ERROR it is undefined behavior to use this value
+//~^ ERROR encountered dangling pointer in final constant
```

Comment:
I am not happy about the duplicate error here, but I am also not sure how to best avoid it.

Reply:
The duplication is because validation also errors about dangling things if I'm reading the stderr file correctly. So, we could just not error in interning in case we encounter a dangling reference, and only error for dangling relocations in padding and such.

Reply:
> So, we could just not error in interning in case we encounter a dangling reference, and only error for dangling relocations in padding and such.

The error from interning here ("encountered dangling pointer in final constant") is the same one that the `dangling_raw_ptr` test checks for. I don't know how to remove that error from `dangling-alloc-id-ice` (a dangling reference) without also removing it from `dangling_raw_ptr`.

Reply:
oooh, these are already leftovers. I don't get how the above can cause the reference the be in the leftover set. Rereading the interner logic, will update once I know.

Reply:
The interned just collects *everything* in the leftover set. Typed interning ignores dangling pointers. Thus, dangling references end up in the same bucket as dangling raw pointers.

To avoid this, we'd need to... maybe skip leftover interning in case we find a dangling reference during the typed phase? Or at least skip emitting that error.

Reply:
seems an aweful lot of complexity just to remove a diagnostic duplication... Let's ignore it?

Reply:
I'm okay with ignoring it for now.